data/net/minecraft/world/level/SignalGetter.mapping

```diff
@@ -1,19 +1,50 @@
 CLASS net/minecraft/world/level/SignalGetter
 	METHOD getBestNeighborSignal (Lnet/minecraft/core/BlockPos;)I
+		COMMENT Returns the highest redstone signal the given position receives from neighboring blocks.
 		ARG 1 pos
 	METHOD getControlInputSignal (Lnet/minecraft/core/BlockPos;Lnet/minecraft/core/Direction;Z)I
+		COMMENT Returns the control signal emitted from the given position in the given direction.
```

Comment:
what's a `control signal`?

Reply:
Something Mojang made up. I just took it from the method name and clarified it more in the rest of the comment. It can return specific types of redstone signals from specific blocks.

Reply:
Eh, Mojang do what Mojang do (or did).

While I can't for certain conclude it, I'm getting the feeling (based on some quick looks on calls to the method) that this is related to the input signals that can be fed to comparators. I'm guessing their definition of an 'control signal' is a signal which can control the behavior of some redstone-emitting block, such as a comparator (diode) block, but that's probably something to investigate for another time.

Reply:
iirc the method is used by repeaters/comparators for their side input, and also the calibrated sculk sensor, to make it listen to specific "frequencies" - the question is always *is it about what the method is used for, or what it does?* and in this case, because the method is surrounded by other "generic" signal-getter methods, I opted to go for the latter.